src/frequenz/client/electricity_trading/_client.py

```diff
@@ -280,6 +286,60 @@ async def stream_public_trades(
                 raise e
         return self._public_trades_streams[public_trade_filter].new_receiver()
 
+    def validate_params(  # pylint: disable=too-many-arguments
+        self,
+        price: Price | None | _Sentinel = NO_VALUE,
+        quantity: Energy | None | _Sentinel = NO_VALUE,
+        stop_price: Price | None | _Sentinel = NO_VALUE,
+        peak_price_delta: Price | None | _Sentinel = NO_VALUE,
+        display_quantity: Energy | None | _Sentinel = NO_VALUE,
+        delivery_period: DeliveryPeriod | None = None,
+        valid_until: datetime | None | _Sentinel = NO_VALUE,
+    ) -> None:
+        """
+        Validate the parameters of an order.
+
+        This method ensures the following:
+        - Price and quantity values have the correct number of decimal places and are positive.
+        - The delivery_start and valid_until values are in the future.
+
+        Args:
+            price: The price of the order.
+            quantity: The quantity of the order.
+            stop_price: The stop price of the order.
+            peak_price_delta: The peak price delta of the order.
+            display_quantity: The display quantity of the order.
+            delivery_period: The delivery period of the order.
+            valid_until: The valid until of the order.
+
+        Raises:
+            ValueError: If the parameters are invalid.
+        """
+        if not isinstance(price, _Sentinel) and price is not None:
+            validate_decimal_places(price.amount, PRECISION_DECIMAL_PRICE, "price")
+        if not isinstance(quantity, _Sentinel) and quantity is not None:
+            validate_decimal_places(
+                quantity.mwh, PRECISION_DECIMAL_QUANTITY, "quantity"
+            )
+        if not isinstance(stop_price, _Sentinel) and stop_price is not None:
+            validate_decimal_places(
+                stop_price.amount, PRECISION_DECIMAL_PRICE, "stop price"
+            )
+        if not isinstance(peak_price_delta, _Sentinel) and peak_price_delta is not None:
+            validate_decimal_places(
+                peak_price_delta.amount, PRECISION_DECIMAL_PRICE, "peak price delta"
+            )
+        if not isinstance(display_quantity, _Sentinel) and display_quantity is not None:
+            validate_decimal_places(
+                display_quantity.mwh, PRECISION_DECIMAL_QUANTITY, "display quantity"
+            )
+        if not isinstance(delivery_period, _Sentinel) and delivery_period is not None:
```

Comment:
The `delivery_period` is not defined as `_Sentinel` type in the function signature. So either the _Sentinel check is redundant here or the function signature is wrong

Reply:
Oh yea good call that's a typo it should never be `Sentinel`